src/rpc/net.cpp

```diff
@@ -303,23 +295,21 @@ UniValue getaddednodeinfo(const UniValue& params, bool fHelp)
             + HelpExampleRpc("getaddednodeinfo", "true, \"192.168.0.201\"")
         );
 
+    if(!g_connman)
+        throw JSONRPCError(RPC_CLIENT_P2P_DISABLED, "Error: Peer-to-peer functionality missing or disabled");
+
     bool fDns = params[0].get_bool();
 
     list<string> laddedNodes(0);
-    if (params.size() == 1)
-    {
-        LOCK(cs_vAddedNodes);
-        BOOST_FOREACH(const std::string& strAddNode, vAddedNodes)
-            laddedNodes.push_back(strAddNode);
-    }
-    else
+    g_connman->GetAddedNodes(laddedNodes);
+
+    if (params.size() > 1)
     {
         string strNode = params[1].get_str();
-        LOCK(cs_vAddedNodes);
-        BOOST_FOREACH(const std::string& strAddNode, vAddedNodes) {
+        BOOST_FOREACH(const std::string& strAddNode, laddedNodes) {
             if (strAddNode == strNode)
             {
-                laddedNodes.push_back(strAddNode);
+                laddedNodes.assign(1, strAddNode);
                 break;
```

Comment:
note: This section is a bit more than just a code move, but it seems to have no functional change.